tests/unit/test_client.py

```diff
@@ -1713,6 +1713,17 @@ def test_client_internal_credential_shim(self):
         )
         self.assertEqual(service_client._get_credentials(), self.credentials)
 
+    def test_unsigned_payload_disables_signing(self):
+        creator = self.create_client_creator()
+        service_client = creator.create_client('myservice', 'us-west-2')
+        test_operation = self.service_description['operations'][
+            'TestOperation'
+        ]
+        test_operation['unsignedPayload'] = True
+        service_client.test_operation(Foo='one')
+        context = self.endpoint.make_request.call_args[0][1]['context']
+        self.assertFalse(context['payload_signing_enabled'])
```

Comment:
Do we have any positive tests enforcing the other side of this?

Reply:
Since this logic moved out of client.py, the test moved to a similar one in handler.py.  I have added an inverse of this test as well